src/main/java/seedu/homerce/model/appointment/Appointment.java

```diff
@@ -114,11 +119,34 @@ public boolean isSame(UniqueListItem other) {
         if (!(other instanceof Appointment)) {
             return false;
         }
-
         Appointment otherAppointment = (Appointment) other;
-        return client.equals(otherAppointment.getClient())
-                && service.equals(otherAppointment.getService())
-                && appointmentDate.equals(otherAppointment.appointmentDate)
-                && timeOfDay.equals(otherAppointment.timeOfDay);
+        if (appointmentDate.equals(otherAppointment.appointmentDate)) {
+            return isClashing(otherAppointment);
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isClashing(Appointment other) {
+        LocalTime startTimeOfThis = getAppointmentTime().getLocalTime();
+        LocalTime startTimeOfOther = other.getAppointmentTime().getLocalTime();
+        if (startTimeOfThis.equals(startTimeOfOther)) {
+            return true;
+        }
+        Duration durationThis = getService().getDuration();
+        Duration durationOther = other.getService().getDuration();
+        LocalTime endTimeOfThis = startTimeOfThis
+            .plusHours((long) Math.floor(durationThis.value))
+            .plusMinutes((long) ((durationThis.value % 1) * 60));
+        LocalTime endTimeOfOther = startTimeOfOther
+            .plusHours((long) Math.floor(durationOther.value))
+            .plusMinutes((long) ((durationOther.value % 1) * 60));
+        System.out.println("Comparing: " + this + " and other: " + other);
+        boolean result = (endTimeOfOther.isAfter(startTimeOfThis) && endTimeOfOther.isBefore(endTimeOfThis))
+                || (endTimeOfOther.isAfter(endTimeOfThis) && startTimeOfOther.isBefore(startTimeOfThis))
+                || (endTimeOfThis.isAfter(endTimeOfOther) && startTimeOfThis.isBefore(startTimeOfOther))
+                || (endTimeOfThis.isAfter(startTimeOfOther) && endTimeOfThis.isBefore(endTimeOfOther));
+        System.out.println("Is there a clash? " + result);
+        return result && appointmentDate.equals(other.appointmentDate);
```

Comment:
The additional `&& appointmentDate.equals(other.appointmentDate)` check is not required right since `isClashing` is only called if `appointmentDate.equals(otherAppointment.appointmentDate)` in line 123

Reply:
Thanks for the catch!